atomic_reactor/util.py

```diff
@@ -891,21 +888,52 @@ def get_inspect_for_image(self, image):
         }
 
         if not blob_config:
-            raise RuntimeError("Image {image_name}: Couldn't get inspect data "
-                               "from digest config".format(image_name=image))
-
-        # set Id, which isn't in config blob
-        # Won't be set for v1,as for that image has to be pulled
-        image_inspect['Id'] = config_digest
+            raise RuntimeError(f"Image {image}: Couldn't get inspect data from digest config")
+        if arch and blob_config['architecture'] != arch:
+            raise RuntimeError(
+                f"Image {image}: Has architecture {blob_config['architecture']}, "
+                f"which does not match specified architecture {arch}"
+            )
+
+        image_inspect = {
+            # set Id, which isn't in config blob
+            # Won't be set for v1,as for that image has to be pulled
+            'Id': config_digest,
+        }
         # only v2 has rootfs, not v1
-        if 'rootfs' in blob_config:
-            image_inspect['RootFS'] = blob_config['rootfs']
+        if rootfs := blob_config.get('rootfs'):
+            image_inspect['RootFS'] = {'Type': rootfs['type'], 'Layers': rootfs['diff_ids']}
 
         for old_key, new_key in config_2_inspect.items():
             image_inspect[new_key] = blob_config[old_key]
 
         return image_inspect
 
+    def _blob_config_by_digest(self, image: ImageName, config_digest: str) -> dict:
+        config_response = query_registry(self._session, image, digest=config_digest, is_blob=True)
+        blob_config = config_response.json()
+        return blob_config
+
+    def _config_and_id_from_manifest_list(
+        self, image: ImageName, manifest_list: dict, arch: Optional[str]
+    ) -> Tuple[dict, str]:
+        # if an architecture was specified, filter the correct architecture
+        filtered_manifests = (
+            m for m in manifest_list["manifests"]
+            if not arch or m.get("platform", {}).get("architecture") == arch
+        )
+        manifest = next(filtered_manifests, None)
+
+        if not manifest:
+            raise RuntimeError(
+                f"Image {image}: No manifest for {arch} architecture in manifest list"
```

Comment:
well you don't know if architecture was even provided

Reply:
+1, None for architecture might be misleading in this error message

Reply:
This can only happen if the manifest list is empty, but technically the [OCI spec](https://github.com/opencontainers/image-spec/blob/main/image-index.md#image-index-property-descriptions) allows that (and the docker spec doesn't say anything on the matter) so I'll add a check for this

Reply:
fixed